onnx/test/data_propagation_test.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from onnx import TensorProto, ONNX_ML
+from onnx.defs import ONNX_DOMAIN, ONNX_ML_DOMAIN, AI_ONNX_PREVIEW_TRAINING_DOMAIN
+from onnx.helper import make_node, make_tensor, make_tensor_value_info
+from shape_inference_test import TestShapeInferenceHelper  # type: ignore
```

Comment:
Yes, at least in my local environment, mypy couldn't resolve it.
```
python setup.py typecheck
...
running typecheck
onnx/test/data_propagation_test.py:8: error: Cannot find implementation or library stub for module named 'shape_inference_test'
onnx/test/data_propagation_test.py:8: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
```

Reply:
Thanks for providing the error. Sorry for changing -- could you please still use `from onnx.test.shape_inference_test import TestShapeInferenceHelper` instead? I did a quick experiment and it seems it can pass mypy typeheck (0.782).

Reply:
Thank you so much for the investigation -- they are really helpful. It's interesting that I cannot reproduce the same error in my local with either `pip install -e .` or `python setup.py`, but the CI failures are there.

I don't have a confident answer about whether adding `__init__.py` will have any side effects or not...  Another possible solution would be using mypy with `--namespace-packages`, which would probably default in future mypy by: https://github.com/python/mypy/issues/8584. However, I did try to run mypy with this config, but it's still throwing the same error (I guess there is an issue about old mypy that `--namespace-packages` does not work for some cases: https://github.com/python/mypy/issues/8584). ONNX is using mypy==0.782, which is at little old.

Since I don't want to block you due to an old mypy issue in ONNX, to be safe without adding `__init__.py`, could you please still use `from shape_inference_test import TestShapeInferenceHelper  # type: ignore` (Sorry for changing my opinion again and again) and please add a TODO comment above saying something like `TODO: remove the following ignore after mypy upgrade in ONNX`. I think we are close to merge this PR. Thank you for the contribution.